meetings/2022-09-28.md

```diff
@@ -0,0 +1,187 @@
+* ...: The other matcher will be satisfied if Bob does not use a client as well.
+* Elf: Could you write the second case context
+
+```ttl
+[]
+  acp:target ex:resourceX ;
+  acp:agent ex:Bob .
+```
+
+* Matthieu: The above context being client-less will resolve successfully if acp:client acp:PublicClient is not required.
+* ...: Really it is just a case of unmitigated resource access. If we say unmitigated resource access exists, then the AuthZ domain needs to be able to express this.
+* Matthieu: Sorry I meant "for acp:AuthenticatedClient", I thought we were discussing unmitigated resource access.
+* Elf: In ACP if you ommit acp:agent in the matcher, any agent (incl. no agent) will match? Assuming other properties in the matcher.
+* Matthieu: An empty matcher is never satisfied.
+* ...: This issue [312](https://github.com/solid/authorization-panel/issues/312) is really just about acknowledging the existence of unmitigated client access.
+* ...: I agree that the "Public" named individuals are a bit misleading because they are not matching on a specific attribute, we could consider creating an always matcher instead...
+* Elf: Ok, I see how AuthenticatedClient makes sense.
+* ...: So as soon as you have a non-empty matcher the attributes targetting a "Public" named individual can be ignored?
+* Matthieu: Yes. And empty matchers are never satisfied.
+* ...: For example, the following Policy will never be satisfied because the empty matcher of the allOf condition will never be satisfied.
+
+```ttl
+[]
+  acp:allow acl:Read ;
+  acp:allOf [ acp:agent ex:Bob ], [] ;
+```
+
+* Matthieu: On the other hand, in the following Policy, the second matcher adds no value. It will be satisfied if Bob is the agent.
+
+```ttl
+[]
+  acp:allow acl:Read ;
+  acp:allOf [ acp:agent ex:Bob ], [ acp:client acp:PublicClient ] ;
+```
+
+* Elf: So in other words, you cannot say that Public is default because of that empty matcher edge case.
+* Matthieu: Indeed.
+* Elf: As long as you have non-empty matcher, any additional attribute with public value could be as well omitted.
+* Matthieu: And the public matcher is always satisfied which is useful to create an allow list for example (see my Public vs Authenticated client example).
+* Elf: I don't know if it would be more consistent to allow things by default in an empty matcher.
+* Matthieu: No because that would be a case of open by default, say if you removed the last attribute of a Matcher, then you would potentially open a Policy by default. To be perfectly honest, some of this should be addressed by specifying an Access Control Resource representation format (disallowing empty matcher/and Policies with no conditions...). But even then, we will want to keep the part of the ACP resolution algorithm enforcing that empty matchers are never satisfied.
```

Comment:
```suggestion
* Matthieu: No, because that would be a case of open by default. Say if you removed the last attribute of a Matcher, then you would potentially open a Policy by default. To be perfectly honest, some of this should be addressed by specifying an Access Control Resource representation format (disallowing empty matcher/and Policies with no conditions...). But even then, we will want to keep the part of the ACP resolution algorithm that enforces that empty matchers are never satisfied.
```